addons/mail/models/mail_activity.py

```diff
@@ -110,6 +110,20 @@ def default_get(self, fields):
             res['res_model_id'] = self.env['ir.model']._get(res['res_model']).id
         return res
 
+    @api.model
+    def _default_activity_type_id(self):
+        ActivityType = self.env["mail.activity.type"]
+        todo = self.env.ref('mail.mail_activity_data_todo', raise_if_not_found=False)
+        current_model_id = self.default_get(['res_model_id', 'res_model'])['res_model_id']
+        if todo and todo.active and (todo.res_model_id.id == current_model_id or not todo.res_model_id):
```

Comment:
technically it is true that people may customize todo ... however not sure we should allow it. Ok, let us say it is valid.